src/Control/SessionHandler/DatabaseSessionHandler.php

```diff
@@ -0,0 +1,222 @@
+     * 3. We don't want sessions to be used in any other ways that DataObjects are often
+     * 4. We only want to build the table if this is the configured save handler
+     */
+    public function requireTable()
+    {
+        $fields = [
+            'ID' => 'PrimaryKey',
+            'SessionID' => 'Varchar(64)',
+            'Expiry' => 'Int',
+            'Data' => 'Text',
+        ];
+        $indexes = [ // @TODO - check this is correct syntax
+            'SessionID' => [
+                'type' => 'unique', // @TODO can I make this the primary key??
+                'columns' => ['SessionID'],
+            ],
+            'Expiry' => [
+                'type' => 'index',
+                'columns' => ['Expiry'],
+            ],
+        ];
+        DB::get_schema()->schemaUpdate(function () use ($fields, $indexes) {
+            DB::require_table(
+                static::config()->get('table_name'),
+                $fields,
+                $indexes,
+                true,
+                DataObject::config()->get('create_table_options')
+            );
+        });
+    }
+
+    /**
+     * Get an SQLSelect for selecting the data for the given session ID.
+     * If $allowExpired is false, expired sessions are explicitly excluded.
+     */
+    private function getSqlSelect(#[SensitiveParameter] string $id, bool $allowExpired = false): SQLSelect
+    {
+        $select = SQLSelect::create(
+            Convert::symbol2sql('Data'),
+            Convert::symbol2sql(static::config()->get('table_name')),
+            [Convert::symbol2sql('SessionID') => $id],
+        );
+        if (!$allowExpired) {
+            $select->addWhere([Convert::symbol2sql('Expiry') . ' >= ?' => DBDatetime::now()->getTimestamp()]);
+        }
+        return $select;
+    }
+
+    /**
+     * Check if a session with this ID exists.
+     * If $allowExpired is false, returns false for expired sessions.
+     */
+    private function sessionExists(#[SensitiveParameter] string $id, bool $allowExpired = false): bool
+    {
+        // Note this is the same logic used in DataQuery::exists()
+        $row = DB::withPrimary(function () use ($id, $allowExpired) {
+            $select = $this->getSqlSelect($id, $allowExpired);
+            $subQuerySql = $select->sql($params);
+            $selectExists = SQLSelect::create('1')->addWhere(['EXISTS (' . $subQuerySql . ')' => $params])->execute();
+            return $selectExists->record();
+        });
+        if ($row) {
+            $result = reset($row);
+        } else {
+            $result = false;
+        }
+        return $result === true || $result === 1 || $result === '1';
+    }
+
+    private function isDatabaseReady()
+    {
+        if (!DB::connection_attempted() || !DB::is_active()) {
+            return false;
+        }
+        return DB::get_schema()->hasTable(static::config()->get('table_name'));
+    }
```

Comment:
This was originally going to use the new `DataObjectSchema::tablesAreReadyForClass()` method, but since we aren't using a `DataObject` (see phpdoc comment for `requireTable()` for reasons why not) that's not suitable.
Ultimately since the schema for this table isn't configurable, the table will either exist and have the correct schema, or not exist.